api/schema/resolvers/PaymentResolver.js

```diff
@@ -77,6 +78,32 @@ module.exports = {
         deletePaymentById: (root, { id }, { models }) => {
             return models.Payment.destroy({ where: { id } })
         },
+        generateBitcoinInvoiceFromPayment: async (root, { paymentId }, { models }) => {
+            
+            const Payment = await models.Payment.findByPk(paymentId)
+            const { amount, client_id, date_paid, external_uuid, external_uuid_type } = Payment.dataValues
+
+            const client = await models.Client.findByPk(client_id)
+            
+            const isClientCurrencyBtc = client.dataValues.currency === 'BTC' || client.dataValues.currency === 'SATS'
+            if (!isClientCurrencyBtc) throw 'Client\'s currency is not Bitcoin'
+
+            const isInvoiceExpired = external_uuid && external_uuid_type === 'bitcoin' && await checkInvoiceExpired(external_uuid)
+            if (external_uuid && date_paid && !isInvoiceExpired) throw 'An active invoice already exists'
+
+            const amountInSats = client.dataValues.currency === 'BTC' ? amount * 1000000 : amount / 100;
+            
+            const newInvoice = await createBitcoinInvoice(amountInSats)
+            return models.Payment.update({
+                external_uuid: newInvoice.id,
+                external_uuid_type: 'bitcoin'
+            }, 
+            { where: 
+                {
+                    id: paymentId
+                } 
+            })
```

Comment:
This notation is a bit wonky, should be
```
return models.Payment.update(
    {
        external_uuid: newInvoice.id,
        external_uuid_type: 'bitcoin'
    },
    {
        where: {
            id: paymentId
        }
    }
)
```

or perhaps
```

return models.Payment.update(
    {
        external_uuid: newInvoice.id,
        external_uuid_type: 'bitcoin'
    },
    {
        where: { id: paymentId }
    }
)
```

this makes it more clear where the parameters of the function start and end